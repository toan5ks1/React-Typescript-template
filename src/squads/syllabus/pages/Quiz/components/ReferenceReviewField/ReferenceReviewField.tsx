import { useMemo } from "react";

import { inferQuery } from "src/squads/syllabus/services/infer-query";

import TextPreview, { TextPreviewProps } from "src/squads/syllabus/components/TextPreview";

export interface ReferenceReviewFieldProps extends Omit<TextPreviewProps, "value"> {
    value: string[];
    label: string;
}

const ReferenceReviewField = (props: ReferenceReviewFieldProps) => {
    const { value, label, ...rest } = props;

    const { data } = inferQuery({
        entity: "learningObjective",
        action: "LO_GET_MANY_BY_IDS",
    })(
        {
            lo_id: value,
        },
        {
            enabled: true,
        }
    );

    const values = useMemo(() => {
        if (!data) return "";

        return data.map((e) => e.name).join(", ");
    }, [data]);

    return <TextPreview {...rest} title={label} value={values} />;
};

export default ReferenceReviewField;
