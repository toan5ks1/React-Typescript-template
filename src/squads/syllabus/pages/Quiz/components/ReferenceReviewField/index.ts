export { default } from "./ReferenceReviewField";
export * from "./ReferenceReviewField";
