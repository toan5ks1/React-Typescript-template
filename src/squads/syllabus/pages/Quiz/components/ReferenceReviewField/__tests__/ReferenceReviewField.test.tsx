import ReferenceReviewField from "../ReferenceReviewField";

import { render, RenderResult, screen } from "@testing-library/react";

const mockListLOs = [
    {
        country: "COUNTRY_NONE",
        created_at: "2021-10-13T07:00:19.764748+00:00",
        display_order: 3,
        grade: 0,
        lo_id: "lo_id",
        name: "LO name",
        school_id: 123,
        study_guide: "",
        subject: "SUBJECT_NONE",
        topic_id: "topic id",
        type: "LEARNING_OBJECTIVE_TYPE_LEARNING",
        updated_at: "2021-10-13T07:00:19.764748+00:00",
    },
];

jest.mock("src/squads/syllabus/services/infer-query", () => {
    return {
        __esModule: true,
        inferQuery: () => () => ({
            data: mockListLOs,
        }),
    };
});

describe("ReferenceReviewField component", () => {
    let wrapper: RenderResult;
    beforeEach(() => {
        wrapper = render(<ReferenceReviewField value={["lo_id"]} label={"Learning objectives"} />);
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should show type of LO is Learning objectives and name of LO correctly", () => {
        expect(screen.getByText("Learning objectives")).toBeInTheDocument();
        expect(screen.getByText(mockListLOs[0].name)).toBeInTheDocument();
    });
});
