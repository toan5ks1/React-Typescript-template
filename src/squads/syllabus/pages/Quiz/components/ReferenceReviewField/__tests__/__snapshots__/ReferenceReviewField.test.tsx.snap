// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferenceReviewField component should match snapshot 1`] = `
.emotion-1 {
  margin-bottom: 8px;
  font-weight: 500;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      Learning objectives
    </div>
    <div
      class="MuiBox-root emotion-0"
    >
      LO name
    </div>
  </div>
</div>
`;
